spec/models/dialog_spec.rb

```diff
@@ -286,6 +286,149 @@
+            'label' => 'updated label',
+            'dialog_groups' => [
+              {
+                'id' => @dialog_group.first.id,
+                'dialog_fields' => [
+                  { 'id' => @dialog_field.first.id }
+                ]
+              }
+            ]
+          },
+          {
+            'label' => 'new tab',
+            'dialog_groups' => [
+              {
+                'label' => 'a new group',
+                'dialog_fields' => [
+                  { 'name' => 'new field', 'label' => 'field' }
+                ]
+              }
+            ]
+          }
+        ]
+      }
+
+      @dialog.update_content(updated_content)
+      expect(@dialog.reload.content.first['dialog_tabs'].count).to eq(2)
+    end
+
+    it 'adds a new dialog group to a tab' do
+      updated_content = {
+        'dialog_tabs' => [
+          {
+            'id' => @dialog_tab.first.id,
+            'label' => 'updated label',
+            'dialog_groups' => [
+              {
+                'id' => @dialog_group.first.id,
+                'dialog_fields' => [
+                  { 'id' => @dialog_field.first.id }
+                ]
+              },
+              {
+                'label' => 'new group',
+                'dialog_fields' => [
+                  { 'name' => 'new field', 'label' => 'field'}
+                ]
+              }
+            ]
+          }
+        ]
+      }
+
+      @dialog.update_content(updated_content)
+      expect(@dialog.reload.content.first['dialog_tabs'].first['dialog_groups'].count).to eq(2)
+    end
+
+    it 'adds a new dialog field to a group' do
+      updated_content = {
+        'dialog_tabs' => [
+          {
+            'id' => @dialog_tab.first.id,
+            'label' => 'updated label',
+            'dialog_groups' => [
+              {
+                'id' => @dialog_group.first.id,
+                'dialog_fields' => [
+                  { 'id' => @dialog_field.first.id },
+                  { 'name' => 'new field', 'label' => 'field' }
+                ]
+              }
+            ]
+          }
+        ]
+      }
+
+      @dialog.update_content(updated_content)
+      expect(@dialog.reload.content.first['dialog_tabs'].first['dialog_groups'].first['dialog_fields'].count).to eq(2)
+    end
+
+    it 'removes a dialog tab from a dialog' do
+      @dialog.dialog_tabs << FactoryGirl.create(:dialog_tab)
+      @dialog.save
+
+      updated_content = {
+        'dialog_tabs' => [
+          {
+            'id' => @dialog_tab.first.id,
+            'dialog_groups' => [
+              {
+                'id' => @dialog_group.first.id,
+                'dialog_fields' => [
+                  { 'id' => @dialog_field.first.id }
+                ]
+              }
+            ]
+          }
+        ]
+      }
+
+      @dialog.update_content(updated_content)
+      expect(@dialog.reload.content.first['dialog_tabs'].count).to eq(1)
+    end
+  end
```

Comment:
Again, would be nice to have two contexts, one for when the 'id' key exists and one when it doesn't.

I also think a lot of this could be re-structured to reuse the "updated_content" portion. My suggestion is something like this:
```
describe "#update_content" do
  let(:updated_content) { {'dialog_tabs' => [{'id' => dialog_tab.first.id, 'dialog_groups' => dialog_groups}]} }
  let(:dialog_tab) # dialog tab setup

  # All your other setup

  context "when the dialog group is X" do
    let(:dialog_groups) # dialog_group setup for X
  end

  context "when the dialog group is Y" do
    let(:dialog_groups) # dialog_group setup for Y
  end
end
```

This structure should be able to show what's different between the tests and why. As it is now, it's difficult to tell the difference between the various `updated_content`s, and without a `context` wrapper, hard to understand why they're different.